### PR TITLE
Provide proper prop-testing with `prop-tests` pro prop-testing package

### PR DESCRIPTION
**Pull request by sandreae**
Implement `prop-tests` tests for creating and encoding operations.

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa)

### CODE REVIEW

**Review by adzialocha — APPROVED** [review]
Cool. Looking forward to use this more in other tests.

And: Why did noone comment on my secret PR re-naming mission? 😎

**Comment by sandreae** [comment]
> And: Why did noone comment on my secret PR re-naming mission? sunglasses

Hahahah, I forgot to comment!! When I first saw it, I thought that's what I actually named the PR, took me a while t realise what had been gooooing on. :rofl: :rofl:

File: p2panda-rs/src/operation/operation.rs [hunk 1]
```diff
@@ -79,7 +81,8 @@ impl Copy for OperationVersion {}
 ///
 /// An action defines the operation format and if this operation creates, updates or deletes a data
 /// document.
-#[derive(Clone, Debug, Eq, PartialEq)]
```
Comment (cafca) [inline p2panda-rs/src/operation/operation.rs:82]: You removed `Eq` here?

File: p2panda-rs/src/tests/prop_tests.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        } else {
+            prop_assert!(operation.is_ok());
+
+            let operation_encoded = OperationEncoded::try_from(operation.as_ref().unwrap()).unwrap();
+            let decoded_operation = Operation::from(&operation_encoded);
+
+            prop_assert_eq!(decoded_operation, operation.unwrap());
+        }
+    }
+}
```
Comment (cafca) [inline p2panda-rs/src/tests/prop_tests.rs:26]: A module docstring could be nice for this, just something short about what a prop test is.

**Review by cafca — CHANGES_REQUESTED** [review]
I have overlooked this PR for too long even when it's actually really really nice. I think a short comment on what `prop_tests.rs` is could be nice in that file.

**Comment by cafca** [comment]
Current problem is that the proptests grow too large with arbitrary hashes being generated and new complexity introduced by `Relation` and friends. I am looking for a way to limit the number of arbitrary hashes that are being tested.

**Comment by sandreae** [comment]
Yeh, I agree. If we find an area where we want to apply it in the future, we can introduce it then.

**Pull request closed without merging**